src/async_buf_read.rs

```diff
@@ -0,0 +1,77 @@
+use pin_project::pin_project;
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::io::{AsyncBufRead, AsyncRead, ReadBuf, Result as IoResult};
+
+pub trait VecAsyncBufReadExt: AsyncRead + AsyncBufRead {
```

Comment:
👍 this is a nice wrapper. What would you think of adding a test or two to this module for it?

Reply:
I have added a test called `test_read_line_without_shutdown()` which is of some use I think. I would have liked to implement `test_read_line_with_shutdown` to test that it stops reading when the shutdown is triggered but I was having difficulty implementing it as I need to both read and write to it. I can give it another go as the review continues if you like.

Reply:
Removed it as it turned out wrappering the inner `AsyncRead` did the job. I did add some tests to `async_read.rs` if you want to keep them.